dwh/query_lib_big_query.py

```diff
@@ -0,0 +1,325 @@
+"""Module that implements Query Engine for BigQuery."""
+import typing as tp
+
+import pandas as pd
+
+from base import EncounterConstraints, ObsConstraints, PatientQuery
+
+
+_DATE_VALUE_SEPARATOR = ","
+
+try:
+    from google.cloud import bigquery
+except ImportError:
+    pass  # not all set up need to have bigquery libraries installed
+
+
+def _build_in_list_with_quotes(values: tp.Iterable[tp.Any]):
+    """
+    Builds the `in` operand for a where clause
+    """
+    return ",".join(('"{}"'.format(x) for x in values))
+
+
+class _BigQueryPatientQuery(PatientQuery):
+    """Concrete implementation of PatientQuery class that serves data stored in
+    BigQuery."""
+
+    def __init__(self, project_name: str, bq_dataset: str, code_system: str):
+        super().__init__(code_system)
+        self._bq_dataset = bq_dataset
+        self._project_name = project_name
+
+    def _build_encounter_query(
+        self,
+        *,
+        bq_dataset: str,
+        base_url: str,
```

Comment:
You should not need this `base_url`, do you? We should remove it from the Spark version (in a separate PR) as [Issue 55](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/issues/55) is already fixed.